src/api/app/controllers/staging_project/staged_requests_controller.rb

```diff
@@ -2,4 +2,53 @@ class StagingProject::StagedRequestsController < StagingProjectController
   def index
     @requests = @project.staged_requests
   end
+
+  # accepts a list of bs_request numbers e.g. 
+  # tosc api -X POST --data "<requests><number>1</number><number>2</number></requests>" '/staging_project/home:Admin:Staging:A/staged_requests'
+  def create
+    authorize @project, :update?
+    # TODO: ugly and can break easily
+    @staging_project = @project
+    @staging_workflow = @project.staging_workflow
+    @project = @staging_workflow.project
+    hash = Xmlhash.parse(request.body.read).with_indifferent_access
+
+    @requests = @staging_workflow.unassigned_requests.where(id: hash[:number])
+
+    # TODO: should happen in background
+    @requests.each do |request|
+      BranchPackage.new(request: request.number, target_project: @staging_project.name, extend_package_names: false).branch # does raise exception
```

Comment:
this will only work for submit requests though - and will not take rings into account for submit requests

Reply:
in another discussion, was said that just submit requests are used in the staging workflow, no @coolo ?

Reply:
creating the source containers in background sounds like a great idea - it's currently a major pain

Reply:
> in another discussion, was said that just submit requests are used in the staging workflow, no

I think I got this wrong then, my fault :cry:. So delete and submit are supported then and there is only one request action, right? We will look into it ...

> creating the source containers in background sounds like a great idea - it's currently a major pain

Good to know. I'm still thinking about what should happen in error case. Should this request then just stay unassigned? Should we paste the a log somewhere? Any good idea? Just brainstorming ...

Reply:
Make sure the dashboard shows the package as not-ready until the background job is done.

Reply:
> Make sure the dashboard shows the package as not-ready until the background job is done.

We are working with the API. It's unrelated with the UI @coolo

Reply:
still the UI needs infos what's going on. And that you need to consider in the model.